drivers/sensor/ina219/ina219.c

```diff
@@ -0,0 +1,378 @@
+		chan_p == SENSOR_CHAN_INA219_BUS_VOLTAGE) {
+
+		rc = ina219_reg_read(dev, INA219_REG_V_BUS, &tmp);
+		if (rc) {
+			LOG_ERR("Error reading bus voltage.");
+			return rc;
+		}
+		data->v_bus = INA219_VBUS_GET(tmp);
+	}
+
+	if (chan == SENSOR_CHAN_ALL ||
+		chan_p == SENSOR_CHAN_INA219_BUS_POWER)	{
+
+		rc = ina219_reg_read(dev, INA219_REG_POWER, &tmp);
+		if (rc) {
+			LOG_ERR("Error reading power register.");
+			return rc;
+		}
+		data->power = tmp;
+	}
+
+	if (chan == SENSOR_CHAN_ALL ||
+		chan_p == SENSOR_CHAN_INA219_BUS_CURRENT) {
+
+		rc = ina219_reg_read(dev, INA219_REG_CURRENT, &tmp);
+		if (rc) {
+			LOG_ERR("Error reading current register.");
+			return rc;
+		}
+		data->current = tmp;
+	}
+
+	return rc;
+}
+
+static int ina219_channel_get(const struct device *dev,
+			      enum sensor_channel chan,
+			      struct sensor_value *val)
+{
+	const struct ina219_config *cfg = dev->config;
+	struct ina219_data *data = dev->data;
+	double tmp;
+	int8_t sign = 1;
+
+	switch ((enum sensor_channel_ina219) chan) {
+	case SENSOR_CHAN_INA219_V_SHUNT:
+		if (INA219_SIGN_BIT(data->v_shunt)) {
+			data->v_shunt = ~data->v_shunt + 1;
+			sign = -1;
+		}
+		tmp = sign * data->v_shunt * INA219_SI_MUL;
+		sensor_value_from_double(val, tmp);
+		break;
+	case SENSOR_CHAN_INA219_BUS_VOLTAGE:
+		tmp = data->v_bus * INA219_V_BUS_MUL;
+		sensor_value_from_double(val, tmp);
+		break;
+	case SENSOR_CHAN_INA219_BUS_POWER:
+		tmp = data->power * cfg->current_lsb * INA219_POWER_MUL * INA219_SI_MUL;
+		sensor_value_from_double(val, tmp);
+		break;
+	case SENSOR_CHAN_INA219_BUS_CURRENT:
+		if (INA219_SIGN_BIT(data->current)) {
+			data->current = ~data->current + 1;
+			sign = -1;
+		}
+		tmp = sign * data->current * cfg->current_lsb * INA219_SI_MUL;
+		sensor_value_from_double(val, tmp);
+		break;
+	default:
+		LOG_DBG("Channel not supported by device!");
+		return -ENOTSUP;
+	}
+
+	return 0;
+}
+
+#ifdef CONFIG_PM_DEVICE
+static int ina219_set_power_state(const struct device *dev,
+				  enum pm_device_state power_state)
+{
+	struct ina219_data *data = dev->data;
+	uint8_t reg_val;
+	int rc;
+
+	if (data->pm_state == power_state) {
+		LOG_DBG("Device already in requested PM_STATE.");
+		return 0;
+	}
+
+	if (power_state == PM_DEVICE_STATE_ACTIVE) {
+		return ina219_init(dev);
+	} else if (power_state == PM_DEVICE_STATE_SUSPEND) {
+		reg_val = INA219_MODE_SLEEP;
+	} else if (power_state == PM_DEVICE_STATE_OFF) {
+		reg_val = INA219_MODE_OFF;
+	} else {
+		return -ENOTSUP;
+	}
+
+	rc = ina219_reg_field_update(dev,
+				INA219_REG_CONF,
+				INA219_MODE_MASK,
+				reg_val);
+	if (rc) {
+		LOG_ERR("Failed to set power state.");
+		return rc;
+	}
+
+	data->pm_state = power_state;
+
+	return 0;
+}
+
+static uint32_t ina219_get_power_state(const struct device *dev,
+		enum pm_device_state *state)
+{
+	struct ina219_data *data = dev->data;
+
+	*state = data->pm_state;
+
+	return 0;
+}
+
+static int ina219_pm_ctrl(const struct device *dev,
```

Comment:
PM code should be adjusted following latest changes, see https://github.com/zephyrproject-rtos/zephyr/pull/36759

Reply:
updated accordingly